src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -78,6 +78,28 @@ export const migrations = {
         delete doc.attributes.savedSearchId;
       }
 
+      // Migrate controls
+      const visStateJSON = get(doc, 'attributes.visState');
+      if (visStateJSON) {
+        try {
+          const visState = JSON.parse(visStateJSON);
+          const controls = get(visState, 'params.controls') || [];
+          controls.forEach((control, i) => {
+            if (!control.indexPattern) return;
+            control.indexPatternRefName = `control_${i}_index_pattern`;
+            doc.references.push({
+              name: control.indexPatternRefName,
+              type: 'index-pattern',
+              id: control.indexPattern,
+            });
+            delete control.indexPattern;
+          });
+          doc.attributes.visState = JSON.stringify(visState);
+        } catch (e) {
+          // Let it go, the data is invalid and we'll leave it as is
```

Comment:
I believe this is intended to catch the `JSON.parse()` error, but since it's got so much logic in the `try{}` the error could be from a number of things and might cause the doc to be partially updated. Would you mind wrapping as little as possible in the `try {} catch {}`?